config/crd/bases/operator.tigera.io_logcollectors.yaml

```diff
@@ -109,6 +109,29 @@ spec:
                       endpoint:
                         description: 'Location of the syslog server. example: tcp://1.2.3.4:601'
                         type: string
+                      logTypes:
+                        description: LogTypes contains a list of types of logs to
+                          export to syslog. By default, if this field is omitted,
+                          it will be set to include all possible values.
+                        items:
+                          description: SyslogLogType represents the allowable log
+                            types for syslog. Allowable values are AuditEE, AuditKube,
+                            DNS, Flows and IDSEvents. * AuditEE corresponds to audit
+                            logs for Enterprise API custom resources. * AuditKube
+                            corresponds to audit logs for Kubernetes API resources.
+                            * DNS corresponds to DNS logs generated by Calico node.
+                            * Flows corresponds to flow logs generated by Calico node.
+                            * IDSEvents corresponds to event logs for the intrusion
+                            detection system (anomaly detection, suspicious IPs, suspicious
+                            domains and global alerts).
+                          enum:
+                          - AuditEE
```

Comment:
Beware of abbreviations in our API please!

Maybe this?

AuditEE -> Audit
AuditKube -> AuditKubernetes

Reply:
> AuditEE -> Audit
AuditKube -> AuditKubernetes

Hmm, yeah I guess the `AuditKube` isn't a good one. Will change to `AuditKubernetes`. 

As for `AuditEE`, don't we tend to attach Enterprise to things that apply to CaliEnt? I guess `AuditEnterprise` isn't necessary better? I wish we had a consistent name for things that apply to CaliEnt that could be attached to things related to CaliEnt.

Reply:
> don't we tend to attach Enterprise to things that apply to CaliEnt? 

I don't know that we do in user-facing situations. My thinking was really predicated on a couple of things:
- "AuditEnterprise" isn't a well-known pattern or standard. It's not immediately obvious to all users what this means.
- This API object is already scoped within Calico Enterprise. We don't need to reiterate to the user that this configures Calico Enterprise, because if they are looking at this object they are already trying to configure Calico Enterprise.

Maybe, could you explain the difference between AuditKube and Audit?

Reply:
From what I remember, I think basically AuditKube is for Kube API server handled resources and AuditEnterprise is for Tigera API server handled resources.

Reply:
Isn't that an implementation detail though? Do we expect all users to know which APIs are handled by which components in the system? 

Maybe a subtle distinction, but I think we need to classify these things by how a user would see them rather than an engineer. e.g., "APIs provided by Calico Enterprise" and "APIs provided by Kubernetes".

Alternatively, we go more flexible and we let the user specify the API groups / versions / kinds that they want audited.

Reply:
> From what I remember, I think basically AuditKube is for Kube API server handled resources and AuditEnterprise is for Tigera API server handled resources.

Yes, that's correct.

> Isn't that an implementation detail though? Do we expect all users to know which APIs are handled by which components in the system?

It was something that came out of discussing with Product what options we wanted to provide the user to control what gets forwarded to Syslog. It was the maximum level of granularity, but I can also see how one can argue that audit logs are audit logs, regardless of whether it's K8s resources or our own custom resources. 

I can double check with Phil on whether there was a specific customer case that drove needing to control the split between two types of audit logs.

Reply:
Just head back from Product. There's no real need to separate the two kinds of audit logs. 

I'll change it so that they are both controlled by a single user-facing option. 

I'll go with `Audit` as the log type option.